### ISSUE TITLE
Pseudo abort, inode mismatch

### ISSUE THREAD

**Issue by RaGrCobham** [priv-kweihmann/meta-sca#9132]
**Describe the bug**
Our custom cmake app recipe encounters a inode mismatch for a header file and bitbake throws pseudo abort even though NOTHING touches the build environment. We aren't changing anything with the package.

The pseudo error goes away if we clean the recipe and run static analysis.

The recipe is excluded from running any sca tool and added path in SCA_FILE_FILTER.

The pseudo error happens during do_install step consistently when static analysis is run AFTER a clean build.

**Used layer**
meta                 
meta-poky            = "HEAD:c40ac16d79026169639f47be76a3f7b9d8b5178e"
meta-freescale       = "HEAD:84137278733326032ab96a41d4f9624513a21cb2"
meta-sca-minified    = "HEAD:dcb65933a5d3b80380bc93221e8330a00905fa5a"

**Additional**
Any ideas on what could be going on?

**Comment by RaGrCobham** [comment]
Yes, SCA_ENABLE 0 and then 1 because that's how our Jenkins pipeline does it. But if we run only one of them with no mixing, it's fine.

> What I'm really wondering about is what the purpose of pseudo in do_install is.

In any case, we're not calling it manually. If I understood correctly, it's to set the correct permissions?

> Is it just this recipe or random ones?
This one, I can only think that compared to other cmake recipes, this one has include files in a separate folder and are included like 'folder/header.h' in 'src' files and just 'header.h' in the header files. The others have them in the same 'src' folder.

**Comment by RaGrCobham** [comment]
I believe I found the issue, not related to SCA (but still not sure why it's there it was triggered). It's really unusual behaviour, we have the app tree as:

`
app
--common
----include
--client
----src
----CMakeLists.txt
--server
----src
`

So CMake builds the app using the client src and common includes. Bitbake had client and common directories in SRC_URI and 'S' variable was overwritten to 'app/client' instead of default 'app'.

S being overwritten caused the issue since it didn't technically include 'app/common', even though the common folder was part of the final package and during runtime there were no complaints.

I created an 'app/CMakeLists.txt' and included client directory, also changed the CMake path references inside there.
In the bitbake recipe, I removed the overwritten S variable. I have yet to get the Pseudo abort again.
